css/src/lib.rs

```diff
@@ -398,10 +325,38 @@ impl TryFrom<&[u8]> for Signature {
     }
 }
 
-fn take(count: usize) -> impl Fn(&[u8]) -> (&[u8], &[u8]) {
+fn take<const SIZE: usize>() -> impl Fn(&[u8]) -> (&[u8], [u8; SIZE]) {
     move |input| {
-        nom::bytes::complete::take::<usize, &[u8], nom::error::Error<&[u8]>>(count)(input)
-            .expect("Size of stream should have been guaranteed to hold the bytes")
+        let parsing =
+            nom::bytes::complete::take::<usize, &[u8], nom::error::Error<&[u8]>>(SIZE)(input)
+                .expect("Size of stream should have been guaranteed to hold the bytes");
+        let array = <[u8; SIZE]>::try_from(parsing.1)
+            .expect("Size of stream should have been guaranteed to hold the bytes");
+        (parsing.0, array)
+    }
+}
+
+impl Display for Signature {
```

Comment:
implementing display trait allows other clients to utilize the pretty printing, it also localizes the error handling as well as ensures consistency.  The previous implementation in main had some extra trailing spaces and forgot a `:` in one of the `name: value` lines.

Reply:
I feel like this should just be the `Debug` implementation, and if we're going to do a `Display` implementation it should be something that's actually human readable, e.g.:

```
Enclave Signature:
       Measurement: 0x{MRENCLAVE}
      Enclave Type: {vendor}
        Product ID: {isvprod}
  Security Version: {svn}
         Signed By: 0x{MRSIGNER}
         Signed On: {strftime(date)}
        Attributes: Debug (Insecure), XFRM, ...
```

Reply:
Additionally, we should unit test this string using the `valid.css` file (i.e. for regressions)

Reply:
Converted to debug implementation and added unit test against valid.css